### PR TITLE
feat(releases): List archived artifacts in files endpoint

### PR DESCRIPTION
**Pull request by jjbayer**
- [x] List both individual and archived files in `/files/` endpoint.
- [x] Downloadable archived artifacts
- [x] `DELETE` archived artifacts  
- [x] Disallow `PUT` for archived artifacts in release file details endpoint.
- [x] Apply changes to org endpoint (which is almost a copy of the project release files endpoint)

This PR requires:
* https://github.com/getsentry/sentry/pull/26729
* https://github.com/getsentry/sentry/pull/26717

### CODE REVIEW

File: src/sentry/api/endpoints/project_release_file_details.py [hunk 1]
```diff
@@ -35,22 +60,77 @@ def download(releasefile):
...
+    def download_from_archive(release, entry):
+        archive_ident = entry["archive_ident"]
+
+        # Do not use ReleaseFileCache here, we view download as a singular event
+        archive_file = ReleaseFile.objects.get(release=release, ident=archive_ident)
+        archive = ZipFile(archive_file.file.getfile())
+        fp = archive.open(entry["filename"])
+        headers = entry.get("headers", {})
+
+        response = StreamingHttpResponse(
```
Comment (untitaker) [inline src/sentry/api/endpoints/project_release_file_details.py:73]: Why not `HttpResponse(fp)`? This would auto-close afaict

File: src/sentry/api/endpoints/project_release_files.py [hunk 2]
```diff
@@ -130,6 +152,73 @@ def post_releasefile(request, release, logger):
         return Response(serialize(releasefile, request.user), status=201)
 
 
+class ListQuerySet:
```
Comment (untitaker) [inline src/sentry/api/endpoints/project_release_files.py:155]: I think it would be better to write a custom paginator before doing this

File: src/sentry/templates/sentry/error-page-embed.js [hunk 3]
```diff
@@ -13,9 +13,9 @@
     };
    */
 
-  var strings = /*{{ strings }}*/ '';
-  var template = /*{{ template }}*/ '';
-  var endpoint = /*{{ endpoint }}*/ '';
+  var strings = /* {{ strings }} */ '';
```
Comment (untitaker) [inline src/sentry/templates/sentry/error-page-embed.js:16]: spurious change?

Comment (jjbayer) [reply src/sentry/templates/sentry/error-page-embed.js]: Must have been the autoformatter, but don't know what triggered it. Reverted this change just to be safe.

File: src/sentry/api/endpoints/project_release_files.py (hunk 2 above)
Comment (jjbayer) [reply src/sentry/api/endpoints/project_release_files.py]: Turns out `ChainPaginator` already does what I need, using that now.

File: src/sentry/api/endpoints/project_release_file_details.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/api/endpoints/project_release_file_details.py]: I went with [FileResponse](https://docs.djangoproject.com/en/1.11/ref/request-response/#fileresponse-objects), which is a subclass of `StreamingHttpResponse`.

File: src/sentry/api/endpoints/project_release_files.py [hunk 4]
```diff
@@ -152,45 +154,34 @@ def post_releasefile(request, release, logger):
...
+        return len(self._files)
 
-        return files[0]
+    def __getitem__(self, range):
+        self._ensure_ready()
 
-    def annotate(self, **kwargs):
-        if kwargs:
-            raise NotImplementedError
+        return [pseudo_releasefile(url, info, self._dist) for url, info in self._files[range]]
```
Comment (untitaker) [inline src/sentry/api/endpoints/project_release_files.py:172]: would do `_ensure_ready` straight in `__init__` or make the sorted list a cached_property, this is hard to follow. for a second I thought you'd index into `dict.items()`

**Review by untitaker — APPROVED** [review]

File: src/sentry/api/endpoints/project_release_files.py (hunk 4 above)
Comment (jjbayer) [reply src/sentry/api/endpoints/project_release_files.py]: Done.

**Review by untitaker — APPROVED** [review]

**Review by jan-auer — APPROVED** [review]
~@beezz can you verify that we have the ability to cache files on web workers, and that cleanup scripts etc are all configured correctly? ~ Edit: I misread, we don't actually use the releasefile cache.
